index.php

```diff
@@ -1,8 +1,27 @@
 <?php
 require_once 'BookDAO.php';
+require_once 'AuthorDAO.php';
+require_once 'GenreDAO.php';
 require_once 'displayFunction.php';
 
 $bookDAO = new BookDAO();
+$authorDAO = new AuthorDAO();
+$genreDAO = new GenreDAO();
+
+
+if (isset($_POST['title'])) {
+    $new = new Book(
+        is_string($_POST['title']) ? $_POST['title'] : false,
+        is_int($_POST['author']) ? $_POST['author']: false,
+        is_int($_POST['genre']) ? $_POST['genre'] : false,
+        is_int($_POST['year']) ? $_POST['year'] : false,
+        is_int($_POST['progressperc']) ? $_POST['progressperc'] : false,
+        is_int($_POST['rating10']) ? $_POST['rating10'] : false,
+        sanitiseUrl($_POST['coverlink']),
+        sanitiseUrl($_POST['gr-link'])
+    );
```

Comment:
You might want a try catch here - if the wrong type does get provided, and this tries to pass false into a book, it will probs throw an exception. Would be better to catch it at this point and display some kind of generic error message